lib/ramble/ramble/spack_runner.py

```diff
@@ -602,16 +602,20 @@ def inventory_hash(self, require_exist=False):
         contents_to_hash = None
 
         if not self.dry_run:
-            if not lock_exists and (require_exist or len(self.env_contents) == 0):
+            if not lock_exists and require_exist and len(self.env_contents) > 0:
                 logger.die(
                     'spack.lock file does not exist in environment '
                     f'{self.env_path}\n'
                     'Make sure your workspace is fully setup with\n'
                     '    ramble workspace setup'
                 )
+            elif len(self.env_contents) == 0:
+                contents_to_hash = {}
             elif lock_exists:
                 with open(spack_file, 'r') as f:
                     contents_to_hash = sjson.load(f)
+            else:
```

Comment:
I don't think we want an error here. I can't remember exactly why, but I think we want to support falling through this in the case that the lock file doesn't exist, but we also don't require it.

Or, maybe this should just be a return in that case, instead of an error?

Reply:
Will update